### PR TITLE
fixes arnold pizza not checking for TRAIT_NODISMEMBER

### LINKED ISSUE #52756: Golems are Immune to the Arm Rippies of Arnold Pizza. (also you can rename arnold pizza)
Golems can eat Arnold Pizza and be perfectly fine! This is criminal! At least, I think so anyway.

Also Arnold Pizza can be renamed and stuff. I think there was a big deal made about it being easy to identify.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #52756

:cl: ShizCalev
fix: Fixed Arnold pizza not checking for TRAIT_NODISMEMBER, resulting in a message being sent to the user even if they were fine after eating it.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Does this mean that I don't have to make this change in my arnold pizza mushroom PR as well?

File: code/modules/food_and_drinks/food/snacks_pizza.dm [hunk 1]
```diff
@@ -179,15 +179,15 @@
 	value = FOOD_ILLEGAL
 
 /obj/item/reagent_containers/food/snacks/proc/try_break_off(mob/living/M, mob/living/user) //maybe i give you a pizza maybe i break off your arm
+	if(prob(50) || (M != user) || !iscarbon(user) || HAS_TRAIT(user, TRAIT_NODISMEMBER))
```
Comment (Rohesie) [inline code/modules/food_and_drinks/food/snacks_pizza.dm:182]:
```suggestion
	if(M != user || !iscarbon(user) || HAS_TRAIT(user, TRAIT_NODISMEMBER) || prob(50))
```

**Comment by Rohesie** [comment]
Travis angery

File: code/modules/food_and_drinks/food/snacks_pizza.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:182]: Is this a change you'd rather not have?

Comment (ShizCalev) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:182]: Is it more performant to run all those checks before the prob check?

Comment (Rohesie) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:182]: Might be a micro-optimization, sure.

**Pull request merged**